Add failsafe to items.phtml

samgranger opened the pull request:
### Description
We received a warning that $exist is undefined - not entirely sure what the exact scenario was but adding this as a failsafe.

I have made sure that $exist is defined (null) in the 'default' and 'other' case.

### Contribution checklist
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

samgranger committed:
Add failsafe to items.phtml

We got a warning that $exist is undefined - not entirely sure what the exact scenario was but adding this as a failsafe.

Added as default fallback and 'other'.

dmanners commented on app/code/Magento/Catalog/view/frontend/templates/product/list/items.phtml:
```diff
@@ -147,7 +147,11 @@ switch ($type = $block->getType()) {
     break;
+        
+    default:
+        $exist = null;
```
Rather than adding the default options here what about adding an `isset` check as part of the if wrapping the content display.

samgranger replied:
Changed as requested. Your suggestion is a neater solution.

samgranger committed:
Check if 'exist' variable is set in list.phtml

samgranger reviewed (commented):
I have removed the extra declarations for $exist and have added an isset check to the if statement as requested.

dmanners commented:
@samgranger thank you for the quick update. I will process this pr now.

dmanners committed:
Update the if part of the list.phtml

This should work if the variable is set
And the variable is true or 1 or greater

samgranger committed:
Set default value for $exist variable in list.phtml

samgranger committed:
Remove 'other' case
